### ISSUE TITLE
Workspace root and LSP root should be handled differently

### ISSUE THREAD

**Issue by TotalKrill** [helix-editor/helix#3993]
In many larger projects, there are multiple languages working together, usually with some nice folder structure going on. Sometimes these are in separate git folders, and sometimes they are large monorepositories. Before #3929 this was achievable with a marker folder `.helix/languages.toml` that could specifiy a small projects Language Server (LS) on a directory basis. While the Helix root was specified as the correct path. 

After the change, now the root for the helix editor and the LS are the same, creating issues with folder structure like this:

```
- workspace root 
   - native rust-project
      - Cargo.toml
   - wasm rust project
      - Cargo.toml
   - dockerfile-project
   - .git folder
```

In these cases if opening helix in one of the rust projects folders, the root will be `workspace root`, initiating the LSP where there are no Cargo.toml file and thus the LS will fail. Creating a Cargo.toml workspace could have been a solution, but in that case, the different targets like wasm/native can start to mess things up, and cargo workspaces does not handle this. 

Enabling the LSP to define where it is started by a marker folder as before would help. 

related isses that probably would be solved by allowing this: 
- https://github.com/helix-editor/helix/issues/3702
- https://github.com/helix-editor/helix/issues/2732
- https://github.com/helix-editor/helix/issues/2970

**Comment by kirawi** [comment]
As mentioned in Matrix, adding a `root-override` field to the `languages.toml` could be a solution.

**Comment by ath3** [comment]
If understand the problem correctly, pr #3553 should fix this issue.

**Comment by the-mikedavis** [comment]
Cases where this wasn't already fixed by #3553 should be fixed by https://github.com/helix-editor/helix/pull/4439
